ts/src/pro/coincheck.ts

```diff
@@ -0,0 +1,208 @@
+
+//  ---------------------------------------------------------------------------
+
+import coincheckRest from '../coincheck.js';
+import { AuthenticationError } from '../base/errors.js';
+import type { Int, Market, OrderBook, Trade } from '../base/types.js';
+import Client from '../base/ws/Client.js';
+import { ArrayCache } from '../base/ws/Cache.js';
+
+//  ---------------------------------------------------------------------------
+
+export default class coincheck extends coincheckRest {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'has': {
+                'ws': true,
+                'watchOrderBook': true,
+                'watchOrders': false,
+                'watchTrades': true,
+                'watchOHLCV': false,
+                'watchTicker': false,
+                'watchTickers': false,
+            },
+            'urls': {
+                'api': {
+                    'ws': 'wss://ws-api.coincheck.com/',
+                },
+            },
+            'options': {
+                'expiresIn': '',
+                'userId': '',
+                'wsSessionToken': '',
+                'watchOrderBook': {
+                    'snapshotDelay': 6,
+                    'snapshotMaxRetries': 3,
+                },
+                'tradesLimit': 1000,
+                'OHLCVLimit': 1000,
+            },
+            'exceptions': {
+                'exact': {
+                    '4009': AuthenticationError,
+                },
+            },
+        });
+    }
+
+    async watchOrderBook (symbol: string, limit: Int = undefined, params = {}): Promise<OrderBook> {
+        /**
+         * @method
+         * @name coincheck#watchOrderBook
+         * @description watches information on open orders with bid (buy) and ask (sell) prices, volumes and other data
+         * @see https://coincheck.com/documents/exchange/api#websocket-order-book
+         * @param {string} symbol unified symbol of the market to fetch the order book for
+         * @param {int} [limit] the maximum amount of order book entries to return
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/#/?id=order-book-structure} indexed by market symbols
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const messageHash = 'orderbook:' + symbol;
```

Comment:
@sc0Vu since the user might provide the unified-id as symbol, we also need to parse it or use `market['symbol']` instead